lib/tty/spinner.rb

```diff
@@ -253,12 +253,10 @@ def auto_spin
             spin
             sleep(sleep_time)
           end
+        ensure
+          write(TTY::Cursor.show, false) if @hide_cursor
         end
       end
-    ensure
-      if @hide_cursor
-        write(TTY::Cursor.show, false)
-      end
```

Comment:
I'd consider having this as well. We hide the cursor before invoking the thread.

Reply:
But that's exactly the issue I tried to fix :)